app/assets/javascripts/components/story/StoryItem.js

```diff
@@ -3,18 +3,32 @@ import PropTypes from 'prop-types';
 import CollapsedStory from './CollapsedStory';
 import ExpandedStory from './ExpandedStory';
 import { connect } from 'react-redux';
-import { toggleStory } from '../../actions/story';
+import { toggleStory, toggleEpic } from '../../actions/story';
 import { releaseIsLate, isHighlighted, isAccepted } from '../../models/beta/story';
 import classNames from 'classnames';
 
-export const StoryItem = ({ story, toggleStory, from, index, sprintIndex, columnId}) => {
+export const StoryItem = ({
+  story,
+  toggleStory,
+  from,
+  index,
+  sprintIndex,
+  columnId,
+  toggleEpic
+}) => {
   const className = classNames({
     'Story--late-release': releaseIsLate(story),
     'Story--highlighted': isHighlighted(story),
     'Story--accepted': isAccepted(story),
   });
   const title = releaseIsLate(story) ? I18n.t('story.warnings.backlogged_release') : '';
 
+  const handleClickLabel = (e, label) => {
+    e.stopPropagation();
+
+    toggleEpic(label, story.projectId);
```

Comment:
I think  `toggleEpic` is not the best name here since it doesn't just toggle it visually, it makes an ajax request and other things, what do you think about calling it `fetchEpic`? Also let's get the projectId from the thunk's `getState()` instead of relying on data coming from the component for that, let's pass just the label to the action.